src/core/qgspointlocator.cpp

```diff
@@ -1166,13 +1166,13 @@ void QgsPointLocator::onFeatureAdded( QgsFeatureId fid )
   }
 
   QgsFeature f;
-  if ( mLayer->getFeatures( QgsFeatureRequest( fid ) ).nextFeature( f ) )
+  if ( mContext )
   {
-    if ( !f.hasGeometry() )
-      return;
-
-    if ( mContext )
+    if ( mLayer->getFeatures( QgsFeatureRequest( fid ) ).nextFeature( f ) )
     {
+      if ( !f.hasGeometry() )
+        return;
+
```

Comment:
I understand you don't want to retrieve the attributes when there is no rendering context which make sense, but could we gather the block getFeatures and hasGeometry to avoid copying-pasting code and then proceed with the existing code?

Reply:
I have tried to keep this PR as simple as possible.